Source/DafnyCore/AST/DafnyAst.cs

```diff
@@ -977,6 +977,321 @@ public class Specification<T> : INode, IAttributeBearingDeclaration
     public override IEnumerable<INode> Children => Expressions;
   }
 
+  public abstract class ASTVisitor<VisitorContext> where VisitorContext : IASTVisitorContext {
```

Comment:
I would put this in a separate file, together with `IASTVisitorContext`.

Reply:
Done.